### PR TITLE
Updated page for Compiler Error CS0116 

### LINKED ISSUE #9204: Needs more info
Can this please have the equivalent's solution?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 98513c27-0ac2-43f9-00ac-72a2989876e7
* Version Independent ID: 50dd4d83-9de1-2901-4704-a45836309010
* Content: [Compiler Error CS0116](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-messages/cs0116)
* Content Source: [docs/csharp/language-reference/compiler-messages/cs0116.md](https://github.com/dotnet/docs/blob/master/docs/csharp/language-reference/compiler-messages/cs0116.md)
* Product: **dotnet-csharp**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

### PR DESCRIPTION
**Pull request by jmarkman**
## Summary

I tackled these aspects of the page:

* Updated the code sample

Previously, the sample was extremely bare bones and only covered methods existing out of the scope of a class. I added some fields (three variables and an autoproperty), changed the methods to be more concrete concepts (latitude and longitude coordinates rather than Method and Method2), and included a struct. I also placed comments in the code samples pointing out what would make the compiler grumpy and what wouldn't. I also made everything public - while it isn't a Best Practice™, it'll allow those reading/learning to simply copy-paste the code into their editor without running into any accessibility issues.

As the issue actually requested, I added a version of the code sample that would compile. However, the previous sample didn't have a main method. Neither does this one. Both will trigger CS5001. I'm not sure if the snippet should have a main method, I'll leave that for the review of the PR.

* Expanded on the actual text on the page

Just lifted what @BillWagner said in his comment and applied it to the leading short paragraph. However, I also expanded on what would actually happen if the user pasted the compiler-triggering code example into their editor of choice (stuck with Visual Studio because hey, it's docs.***microsoft***.com, right?). I removed the bit about C/C++ programmers because there are plenty of other languages out there where the C# structure is a foreign concept. Scripting languages like Javascript and Python come to mind.

I also added an inline link to the [General Structure of a C# Program](../../../csharp/programming-guide/inside-a-program/general-structure-of-a-csharp-program.md) article even though it was listed in the **See Also** section.

Fixes #9204

### CODE REVIEW

File: docs/csharp/language-reference/compiler-messages/cs0116.md [hunk 1]
```diff
@@ -9,37 +9,92 @@ ms.assetid: 4cb137b5-ec29-4c1a-adde-9f8424cb9496
 ---
 # Compiler Error CS0116
 
-A namespace cannot directly contain members such as fields or methods  
-  
-A namespace can contain only other namespaces and type declarations. For more information, see the [namespace keyword](../keywords/namespace.md) article.
-
-This error is often generated by developers from a C/C++ background who forget that in C#, methods and variables must be declared and defined within a struct or class.
-  
+A namespace cannot directly contain members such as fields or methods. However, a namespace can contain other namespaces, structs, and classes. For more information, see the [namespace keyword](../keywords/namespace.md) article.
```
[the remark this answers is not in the document]
Comment (BillWagner) [reply docs/csharp/language-reference/compiler-messages/cs0116.md]: When you split these paragraphs, I'd remove "However" from the start of the next paragraph.

File: docs/csharp/language-reference/compiler-messages/cs0116.md [hunk 2]
```diff
@@ -9,37 +9,92 @@ ms.assetid: 4cb137b5-ec29-4c1a-adde-9f8424cb9496
...
+    // Autoproperties also fall under the definition of this rule
+    public string LocationName { get; set; } 
+
+    // This method as well: if it isn't in a class or a struct, it's violating CS0116
+    public void DisplayLatitude()
+    {
+        Console.WriteLine($"Lat: {latitude}");
+    }
+
+    // Even if there's a properly defined class and method or a properly defined struct
```
Comment (BillWagner) [inline docs/csharp/language-reference/compiler-messages/cs0116.md]: I'd remove this comment. I think it distracts from the explanation.

**Review by BillWagner — CHANGES_REQUESTED** [review]
Overall, this is great @jmarkman 

Once you address these comments, it's ready to go.

Thanks for contributing to .NET docs.

File: docs/csharp/language-reference/compiler-messages/cs0116.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (jmarkman) [reply docs/csharp/language-reference/compiler-messages/cs0116.md]: Got it, will fix.

**Review by BillWagner — APPROVED** [review]

**Comment by BillWagner** [comment]
Thanks for making this changes @jmarkman I'll :shipit: now.

Thanks again for contributing to docs.

**Comment by BillWagner** [comment]
Thanks for making this changes @jmarkman I'll :shipit: now.

Thanks again for contributing to docs.

**Comment by BillWagner** [comment]
Thanks for making this changes @jmarkman I'll :shipit: now.

Thanks again for contributing to docs.

**Comment by BillWagner** [comment]
Thanks for making this changes @jmarkman I'll :shipit: now.

Thanks again for contributing to docs.

**Comment by BillWagner** [comment]
Thanks for making this changes @jmarkman I'll :shipit: now.

Thanks again for contributing to docs.

**Pull request merged**